### PR TITLE
Terminate trade on Insufficient balance

### PR DESCRIPTION
**Pull request by gokula-krishna-dev**

### CODE REVIEW

File: src/botPage/bot/Interpreter.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ export default class Interpreter {
     init() {
         this.$scope = createScope();
         this.bot = new Interface(this.$scope);
+        this.blockly = new _Blockly();
```
Comment (aminmarashi) [inline src/botPage/bot/Interpreter.js]: You don't need to import blockly, instead, you can use `this.stop()` to stop the interpreter, which in turn stops the UI. Look at [this](https://github.com/binary-com/binary-bot/pull/452/files#diff-a01567c526c81e4c69f7321eda89e0faL92)

File: src/botPage/bot/Interpreter.js [hunk 2]
```diff
@@ -90,11 +92,16 @@ export default class Interpreter {
                 globalObserver.emit('Error', e);
                 const { initArgs, tradeOptions } = this.bot.tradeEngine;
                 this.stop();
-                this.init();
-                this.$scope.observer.register('Error', onError);
-                this.bot.tradeEngine.init(...initArgs);
-                this.bot.tradeEngine.start(tradeOptions);
-                this.revert(this.startState);
+                const unrecoverableErrorList = ['InsufficientBalance'];
```
Comment (aminmarashi) [inline src/botPage/bot/Interpreter.js]: You can move this to https://github.com/gokula-krishna-dev/binary-bot/blob/b62ca069468e692f6f72bc6a023ebb276f240789/src/botPage/bot/Interpreter.js#L88, and add the check for `CustomLimitsReached`.

Comment (aminmarashi) [reply src/botPage/bot/Interpreter.js]: I would prefer an optional error argument for `shouldRestartOnError` instead:

```
const unrecoverableErrorList = ['InsufficientBalance', 'CustomLimitsReached'];

const shouldRestartOnError = (bot, e = {}) => !unrecoverableErrorList.includes(e.name)  && botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;

...

if (!shouldRestartOnError(this.bot, e) || !botStarted(this.bot)) {
  reject(e);
  return;
}

```

File: src/botPage/bot/Interpreter.js [hunk 3]
```diff
@@ -3,9 +3,11 @@ import { observer as globalObserver } from 'binary-common-utils/lib/observer';
 import { createScope } from './CliTools';
 import Interface from './Interface';
 
+const unrecoverableErrorList = ['InsufficientBalance', 'CustomLimitsReached'];
 const botInitialized = bot => bot && bot.tradeEngine.options;
 const botStarted = bot => botInitialized(bot) && bot.tradeEngine.tradeOptions;
-const shouldRestartOnError = bot => botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;
+const shouldRestartOnError = (bot, e = {}) =>
+    unrecoverableErrorList.includes(e.name) && botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;
```
Comment (aminmarashi) [inline src/botPage/bot/Interpreter.js]: I think this should be `!unrecoverableErrorList.includes(e.name)`

File: src/botPage/bot/Interpreter.js [hunk 4]
```diff
@@ -3,9 +3,11 @@ import { observer as globalObserver } from 'binary-common-utils/lib/observer';
 import { createScope } from './CliTools';
 import Interface from './Interface';
 
+const unrecoverableErrors = ['InsufficientBalance', 'CustomLimitsReached'];
 const botInitialized = bot => bot && bot.tradeEngine.options;
 const botStarted = bot => botInitialized(bot) && bot.tradeEngine.tradeOptions;
-const shouldRestartOnError = bot => botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;
+const shouldRestartOnError = (bot, e = {}) =>
+    unrecoverableErrors.includes(e.name) && botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;
```
Comment (aminmarashi) [inline src/botPage/bot/Interpreter.js]: I think this should be `!unrecoverableErrors.includes(e.name) `

Comment (gokula-krishna-dev) [reply src/botPage/bot/Interpreter.js]: If the error is matches the unrecoverableErrors array then it should stop. I have tested for both positive and negative test cases, it seems to be working.

Comment (aminmarashi) [reply src/botPage/bot/Interpreter.js]: But the opposite makes sense to me: If the error is _not_ in `unrecoverableErrors` and bot is initialized and ... then we should restart.

**Pull request closed without merging**